app/src/androidTest/java/com/github/h3lp3rs/h3lp/forum/ForumCacheTest.kt

```diff
@@ -0,0 +1,166 @@
+package com.github.h3lp3rs.h3lp.forum
+
+import androidx.test.core.app.ApplicationProvider
+import com.github.h3lp3rs.h3lp.database.Databases.*
+import com.github.h3lp3rs.h3lp.database.Databases.Companion.setDatabase
+import com.github.h3lp3rs.h3lp.database.MockDatabase
+import com.github.h3lp3rs.h3lp.forum.implementation.CachedForum
+import com.github.h3lp3rs.h3lp.signin.SignInActivity.Companion.globalContext
+import com.github.h3lp3rs.h3lp.storage.Storages.Companion.resetStorage
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class CacheForumTest {
+
+    // Useful variables
+    private lateinit var rawForum: Forum
+    private lateinit var cachedForum: Forum
+
+    @Before
+    fun setup() {
+        globalContext = ApplicationProvider.getApplicationContext()
+        resetStorage()
+        setDatabase(PREFERENCES, MockDatabase())
+        ForumCategory.mockForum()
+        rawForum = ForumCategory.root().child(ForumCategory.GENERAL.name)
+        cachedForum = CachedForum(rawForum)
+    }
+
+    private fun resetOnlineForum() {

```

Comment:
I don't think resetting the online forum is the "good test" for the cached forum, for me, you should switch to a forum that only returns failed futures (which is what would happen if the user lost their internet connection)

Reply:
Not really, since as explained above, an empty forum only returns failures.
There are no differences in our implementation between both disconnection and absence of data. 
So the current version is fine.